src/worldstate/routine/Gate/src/alignWithGate.cpp

```diff
@@ -0,0 +1,29 @@
+//
+// Created by joel on 12/05/18.
+//
+#include "alignWithGate.h"
+
+void alignWithGate::setupNodeSubscriptions(ros::NodeHandle nh) {
+    std::string gateDetectTopic = "/gateDetect/output";
+    nh.subscribe(gateDetectTopic, 10, &alignWithGate::gateDetectCallBack, this);
+}
+
+void alignWithGate::gateDetectCallBack(
+const gate_detect::gateDetectMsg::ConstPtr& msg) {
+    worldstate::state_msg msg_to_publish;
+    msg_to_publish.state = worldstate::state_msg::aligningWithGate;
+
+    if (!(msg->detectLeft || msg->detectRight || msg->detectTop))
+        msg_to_publish.state = worldstate::state_msg::locatingGate;
+
+    if (msg->detectLeft && msg->detectRight && msg->detectTop) {
+        double distBtwnHorizontalGates =
+        fabs(msg->distanceLeft - msg->distanceRight);
```

Comment:
In general, we should try and use `std::abs` instead of `fabs`, because `std::abs` is overloaded for all numerical types (and so just works). 

https://stackoverflow.com/questions/3118165/when-do-i-use-fabs-and-when-is-it-sufficient-to-use-stdabs?utm_medium=organic&utm_source=google_rich_qa&utm_campaign=google_rich_qa